src/mlpack/methods/ann/loss_functions/poisson_nll_loss_impl.hpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file poisson_nll_loss_impl.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Implementation of the Poisson NegativeLogLikelihood class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_POISSON_NLL_LOSS_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_POISSON_NLL_LOSS_IMPL_HPP
+
+
+// In case it hasn't yet been included.
+#include "poisson_nll_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+PoissonNLLLoss<InputDataType, OutputDataType>::PoissonNLLLoss(
+  const bool logInput,
+  const bool full,
+  const double eps,
+  const bool reduce):
+  _logInput(logInput),
+  _full(full),
+  _eps(eps),
+  _reduce(reduce)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+double PoissonNLLLoss<InputDataType, OutputDataType>::Forward(
+  const InputType&& input,
+  TargetType&& target)
+{
+  InputType loss(size(input));
+  if (_logInput)
+  {
+    loss = arma::exp(input) - target % input;
+  }
+  else
+  {
+    loss = input - target % arma::log(input + _eps);
+  }
+
+  if (_full)
+  {
+    // Stirling's approximation :
+    // log(n!) = n * log(n) - n + log(2 * pi * n)
+    // Select all elements greater than 1 in target and
+    // add ```target * log(target) - target + log(2 * pi * target)```
+    loss.elem(find(target > 1)) += target * arma::log(target) - target
+                                 + arma::log(2 * arma::datum::pi * target) / 2;
+  }
+
+  if (_reduce)
+  {
+    return arma::accu(loss)/loss.n_elem;
```

Comment:
Missing space around the `/` operator. There are other places where this applies as well.